collections/_posts/2024-01-23-OSS-Net_MemoryEfficientHighResolutionSemanticSegmentationof3DMedicalData.md

```diff
@@ -0,0 +1,181 @@
+tags: deep-learning implicit-neural-representation segmentation memory-efficient
+author: "Maylis Jouvencel"
+cite:
+    authors: "Christoph Reich, Tim Prangemeier, Özdemir Cetin, Heinz Koeppl"
+    title:   "OSS-Net: Memory Efficient High Resolution Semantic Segmentation of 3D Medical Data "
+    venue:   "BMVC 2021"
+pdf: "https://arxiv.org/pdf/2110.10640.pdf"
+---
+
+# Notes
+* Link to the code [here](https://github.com/ChristophReich1996/OSS-Net)
+
+# Highlights
+* The goal of the paper is to perform segmentation using neural implicit functions (NIFs) to avoid memory limitations of 3D CNNs.
+* The authors build up on Occupancy Networks to include advantages from 3D CNNs and NIFs and apply their method to segmentation of tumors.
+
+# Introduction
+
+State-of-the-art in 3D medical data segmentation relies on 3D CNNs that have significant limitations regarding their computation complexity and the memory consumption which grows cubically in memory.
+
+The authors propose to get rid of the voxelized structure and to use NIFs (see previous post [^1] for more details) and focus on Occupancy Networks (ONet) [^2]. 
+
+However, ONet is slow at inference and its expressiveness is limited because it uses a global latent code to represent a shape.
+
+Therefore authors combine a 3D CNN encoder with an ONet decoder to take advantage of the segmentation performance of 3D CNNs and the memory efficiency of ONet.
+
+# Method
+
+**What are Occupancy Networks (ONet)?**
+ONets are networks which learn an occupancy function representing a 3D object. This is described by the mapping:  $$f_{\theta}:\mathbb{R}^3 \times \mathcal{X} \rightarrow [0,1]$$. The inputs are the coordinates of a point $$p\in\mathbb{R}^3$$ and an observation $$x\in\mathcal{X}$$. The output is the occupancy probability $$o\in[0,1]$$ for the point given the observation. This value $$o$$ expresses whether the point $$p$$ is located inside ($$o=1$$) or outside ($$o=0$$) of the continuous object boundary.
+
+
+In OSS-Net, they add a local observation $$z$$, which is a local patch around the point. This is thus described by the mapping: $$f_{\theta}:\mathbb{R}^3 \times \mathcal{X} \times \mathcal{Z} \rightarrow [0,1]$$.
+
+## Architecture
+
+
+
+![](/collections/images/OSSNet/architecture.jpg)
+
+*Figure 1: OSS-Net architecture*
+
+The architecture of OSS-Net includes:
+
+- a **3D CNN encoder**
+    - ResNet-like architecture
+    - input: downscale 3D volume
+    - ouput: global latent vector which consists in the concatenation of the output skip connections (see Fig. 2) 
+    - output: a low resolution segmentation (used in an auxiliary loss and during inference to increase speed) 
+
+![](/collections/images/OSSNet/3DCNN_encoder.jpg)
+*Figure 2: 3D CNN encoder architecture*
+
+
+- a **patch encoder**
+    - consists in two 3D convolution layers
+    - input: $$n$$ patches corresponding to $$n$$ locations in the volume
+    - output: $$n$$ local latent vectors (one for each patch)
```

Comment:
Maybe it would help to connect the outputs of the two encoders with the notations introduced above.